framework/test/jest/unit/specs/modules/chain/bft/bft.spec.js

```diff
@@ -583,6 +583,129 @@ describe('bft', () => {
 			});
 		});
 
+		describe('#isBFTProtocolCompliant', () => {
+			let bft;
+			let blocks;
+
+			beforeEach(async () => {
+				bft = new BFT(bftParams);
+				storageMock.entities.Block.get.mockReturnValue([]);
+				await bft.init();
+				storageMock.entities.Block.get.mockClear();
+
+				// Setup BFT module with blocks
+				const numberOfBlocks = 101;
+				blocks = generateBlocks({
+					startHeight: 1,
+					numberOfBlocks,
+				});
+
+				for (const block of blocks) {
+					await bft.addNewBlock(block);
+				}
+			});
+
+			it('should THROW if block is not provided', async () => {
+				expect(() => bft.isBFTProtocolCompliant()).toThrow(
+					'No block was provided to be verified',
+				);
+			});
+
+			it('should return TRUE when B.maxHeightPreviouslyForged is equal to 0', async () => {
+				const block = {
+					height: 102,
+					generatorPublicKey: 'zxc',
+					maxHeightPreviouslyForged: 0,
+				};
+
+				expect(bft.isBFTProtocolCompliant(block)).toBe(true);
+			});
+
+			it('should return FALSE when B.maxHeightPreviouslyForged is equal to B.height', async () => {
+				const block = {
+					height: 203,
+					maxHeightPreviouslyForged: 203,
+				};
+
+				expect(bft.isBFTProtocolCompliant(block)).toBe(false);
+			});
+
+			it('should return FALSE when B.maxHeightPreviouslyForged is greater than B.height', async () => {
```

Comment:
I think in general the description of tests should reflect the behavior other than the condition. For instance, in this case, we can say `should return FALSE when a delegate does not constructively participate in the Lisk-BFT protocol`. I know this also sounds a bit shallow but I think you could make it even more clear. My point is we should write the behaviors as test descriptions. You may wanna check DPOS.apply for reference. It's also not perfect but would give you the idea. Please just take another look and see if there are ways to improve the descriptions.